tests/e2e/standalone/init_test.go

```diff
@@ -0,0 +1,220 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package standalone
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/dapr/cli/tests/e2e/common"
+	"github.com/dapr/cli/tests/e2e/spawn"
+	"github.com/docker/docker/api/types"
+	dockerClient "github.com/docker/docker/client"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
+)
+
+func TestStandaloneInit(t *testing.T) {
+	// Ensure a clean environment
+	must(cmdUninstall, "failed to uninstall Dapr")
+
+	daprRuntimeVersion, daprDashboardVersion := common.GetVersionsFromEnv(t)
+
+	output, err := cmdInit(daprRuntimeVersion)
+	t.Log(output)
+	require.NoError(t, err, "init failed")
+	assert.Contains(t, output, "Success! Dapr is up and running.")
+
+	homeDir, err := os.UserHomeDir()
+	require.NoError(t, err, "failed to get user home directory")
+
+	daprPath := filepath.Join(homeDir, ".dapr")
+	require.DirExists(t, daprPath, "Directory %s does not exist", daprPath)
+
+	verifyContainers(t, daprRuntimeVersion)
+	verifyBinaries(t, daprPath, daprRuntimeVersion, daprDashboardVersion)
+	verifyConfigs(t, daprPath)
+}
+
+// verifyContainers ensures that the correct containers are up and running.
+// Note, in case of slim installation, the containers are not installed and
+// this test is automatically skipped.
+func verifyContainers(t *testing.T, daprRuntimeVersion string) {
+	t.Run("verifyContainers", func(t *testing.T) {
+		if isSlimMode() {
+			t.Skip("Skipping container verification because of slim installation")
+		}
+
+		cli, err := dockerClient.NewClientWithOpts(dockerClient.FromEnv)
+		require.NoError(t, err)
+
+		containers, err := cli.ContainerList(context.Background(), types.ContainerListOptions{})
+		require.NoError(t, err)
+
+		daprContainers := map[string]string{
+			"dapr_placement": daprRuntimeVersion,
+			"dapr_zipkin":    "",
+			"dapr_redis":     "",
+		}
+
+		for _, container := range containers {
+			t.Logf("Found container: %v %s %s\n", container.Names, container.Image, container.State)
+			if container.State != "running" {
+				continue
+			}
+			name := strings.TrimPrefix(container.Names[0], "/")
+			if expectedVersion, ok := daprContainers[name]; ok {
+				if expectedVersion != "" {
+					versionIndex := strings.LastIndex(container.Image, ":")
+					if versionIndex == -1 {
+						continue
+					}
+					version := container.Image[versionIndex+1:]
+					if version != expectedVersion {
+						continue
+					}
+				}
+
+				delete(daprContainers, name)
+			}
+		}
+
+		assert.Empty(t, daprContainers, "Missing containers: %v", daprContainers)
+	})
+}
+
+// verifyBinaries ensures that the correct binaries are present in the correct path.
+func verifyBinaries(t *testing.T, daprPath, runtimeVersion, dashboardVersion string) {
```

Comment:
on slim mode ... verify placement binary install ...

Reply:
> @shubham1172 how are you guaranteeing the order of execution ie the publish test runs after init etc ... 

I am not. The tests should not be tightly coupled. Today if I make changed to Publish tests, I need to run the entire Standalone test suite. This change decouples the tests. There is a function call in Publish test to ensure that Dapr is installed. This ensures that locally anyone can selectively run tests.


> Also a lot more tests are there in standalone init that needs to be added ...

Yes, this is a WIP PR, will add them shortly.

Reply:
Only containers/components are skipped and the configuration is slightly altered (no zipkin) for slim mode.

Reply:
Sorry I missed this earlier, I was not checking "placement". Updated, thanks!